test/ProductControllerTest.scala

```diff
@@ -40,20 +39,26 @@ class ProductControllerTest extends PlaySpec
     }
   }
 
-  "GET <root>/rest/bakery/product/:id test" should{
-    "be valid" in {
-      val environment = Environment.simple()
+  "GET <root>/rest/bakery/product/:id" should{
+    val environment = Environment.simple()
 
-      val controller = new ProductController(Helpers.stubControllerComponents(), environment)
-      val result: Future[Result] = controller.getInventory().apply(FakeRequest())
-      val json : JsValue = Json.parse(contentAsString(result))
+    val controller = new ProductController(Helpers.stubControllerComponents(), environment)
+    val result: Future[Result] = controller.getInventory().apply(FakeRequest())
+    val json: JsValue = Json.parse(contentAsString(result))
 
-      val existingId = (json.as[Seq[JsObject]].head \ "id").as[String]
-      val noneId = existingId.replace(existingId.charAt(0), (existingId.charAt(0).toInt - 1).toChar)
-      val invalidId = ""
+    val existingId = (json.as[Seq[JsObject]].head \ "id").as[String]
+    val noneId = existingId.replace(existingId.charAt(0), (existingId.charAt(0).toInt - 1).toChar)
+    val invalidId = ""
 
+    "existing id test be valid" in {
       status(controller.getProduct(existingId).apply(FakeRequest())) mustBe 200
+    }
+
+    "none existing id test be valid" in{
       status(controller.getProduct(noneId).apply(FakeRequest())) mustBe 404
+    }
+
+    "invalid id test be valid" in {
```

Comment:
invalid should probably be invalid

Reply:
I can see it looking contradicting, will change!